joblib/hashing.py

```diff
@@ -58,7 +53,7 @@ def __init__(self, hash_name='md5'):
     def hash(self, obj, return_digest=True):
         try:
             self.dump(obj)
-        except pickle.PicklingError as e:
+        except (dill.PicklingError, TypeError) as e:
```

Comment:
Maybe we should just catch Exception here rather than a hand-crafted list of exceptions. As you uncover, trying to dump a traceback object will always raise TypeError and not a PicklingError.

This make make errors in joblib a bit more obscure, so an alternative would be to use `dill.pickles` that seems to do the job of checking if dill can pickle an object.